ndindex/ndindex.py

```diff
@@ -65,6 +66,91 @@ def __init__(self, f):
     def __get__(self, obj, owner):
         return self.f(owner)
 
+class _Guard:
+    def __init__(self):
+        self.val = False
+
+    def __call__(self):
+        if self.val:
+            return True
+        else:
+            self.val = True
+            return False
+
+def parse_index(node_or_string):
+    """
+    "Safely" (needs validation) evaluate an expression node or a string containing
```

Comment:
What does "needs validation" here mean?

Reply:
I guess it depends on what we mean by "safely". `parse_index('('*1000 + ')'*1000)` raises MemoryError (but so does `ast.literal_eval('('*1000 + ')'*1000)`). It shouldn't be possible to execute arbitrary code, though, as far as I can tell.

Reply:
doesn't really mean anything, mostly just a reflection of my paranoia about building a) any parser, and b) something meant to take raw user input. My goal has been to be something that's at least as safe as the existing `literal_eval`. At this point I've examined the code and the surrounding issues enough that I'm reasonably confident this has actually been archived, so I'll edit out the wishy-washy language